src/Resizer.php

```diff
@@ -123,7 +123,23 @@ protected function executeResize(ImageInterface $image, ResizeCoordinates $coord
         // Atomic write operation
         $tmpPath = $this->filesystem->tempnam($dir, 'img');
         $this->filesystem->chmod($tmpPath, 0666, umask());
-        $imagineImage->save($tmpPath, $imagineOptions);
+
+        try {
+            $imagineImage->save($tmpPath, $imagineOptions);
+        } catch (\GmagickException $exception) {
+            if (false === stripos($exception->getMessage(), 'no encode delegate')) {
+                throw $exception;
+            }
+
+            // Fix Gmagick bug “No encode delegate for this image format”, see #67
+            $gmagick = new \Gmagick();
+            $gmagick->newimage(1, 1, (new \GmagickPixel('#FFFFFF'))->getcolor(false));
+            $gmagick->setimageformat(strtoupper($imagineOptions['format']));
+            $gmagick->getImageBlob();
```

Comment:
How does creating a new image fix the issue? And is it correct that we are still using `strtolower()` in line 115?

Reply:
I think Gmagick somehow caches which delegate to use for which format. Rendering an image with the uppercase format once seems to fix the issue for all consecutive calls in the same PHP process.

Reply:
Imagine expects the `format` option to be in lowercase, so the `strtolower()` from 115 should be correct.

Reply:
Can we not use `Imagine\Gmagick\Image::getGmagick()` to retrieve the Gmagick object and call `setimageformat()` on it?

Reply:
I don’t think so. The format would get overwritten when calling `->save()`, and omitting the format is not possible because the quality options would not work anymore.